### ISSUE TITLE
Varnish 6.2.0 - Assert error in h2_rx_data(), http2/cache_http2_proto.c line 775:   Condition((h2->mailcall) == 0) not true. 

### ISSUE THREAD

**Issue by jpastuszek** [varnishcache/varnish-cache#3040]
I am regularly getting this panic with Varnish 6.2.0. 

Looks like 6.2.0 contains the fix https://github.com/rezan/varnish-cache/commit/e5f0fa038cf9f5c54110c9ceda32b56d99255362 but it is not sufficient.

```
Panic at: Tue, 30 Jul 2019 17:48:36 GMT
Assert error in h2_rx_data(), http2/cache_http2_proto.c line 775:
  Condition((h2->mailcall) == 0) not true.
version = varnish-6.2.0 revision b14a3d38dbe918ad50d3838b11aa596f42179b54, vrt api = 9.0
ident = Linux,4.9.70-25.242.amzn1.x86_64,x86_64,-junix,-smalloc,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-hcritbit,epoll
now = 21194017.082950 (mono), 1564508914.654142 (real)
Backtrace:
  0x43c6bb: /usr/sbin/varnishd() [0x43c6bb]
  0x49e102: /usr/sbin/varnishd(VAS_Fail+0x42) [0x49e102]
  0x46b41f: /usr/sbin/varnishd() [0x46b41f]
  0x46dec3: /usr/sbin/varnishd(h2_rxframe+0x463) [0x46dec3]
  0x46ffea: /usr/sbin/varnishd() [0x46ffea]
  0x45b743: /usr/sbin/varnishd() [0x45b743]
  0x45bc00: /usr/sbin/varnishd() [0x45bc00]
  0x7f5f34f61de5: /lib64/libpthread.so.0(+0x7de5) [0x7f5f34f61de5]
  0x7f5f34c8e30d: /lib64/libc.so.6(clone+0x6d) [0x7f5f34c8e30d]
thread = (cache-worker)
pthread.attr = {
  guard = 4096,
  stack_bottom = 0x7f5f2d47d000,
  stack_top = 0x7f5f2d48b000,
  stack_size = 57344,
}
thr.req = 0x7f53fc2f3020 {
  vxid = 564692767, transport = H2
  step = 0x0,
  req_body = R_BODY_INIT,
  err_code = 1, err_reason = (null),
  restarts = 0, esi_level = 0,
  sp = 0x7f53fd552420 {
    fd = 46, vxid = 564692767,
    t_open = 1564508563.103239,
    t_idle = 1564508914.653217,
    ws = 0x7f53fd552460 {
      id = \"ses\",
      {s, f, r, e} = {0x7f53fd552498, +224, (nil), +352},
    },
    transport = H2 {
      streams {
        0x00000000 idle
        0x00000001 c-rem
        0x00000003 c-rem
        0x000000a5 c-rem
        0x000000f9 open
      }
    }
    client = 154.66.183.11 51184 :2443,
  },
  ws = 0x7f53fc2f3170 {
    id = \"req\",
    {s, f, r, e} = {0x7f53fc2fc0b0, +1696, (nil), +61256},
  },
  http_conn = 0x7f53fc2fc050 {
    fd = 46 (@0x7f53fd552444),
    doclose = NULL,
    ws = 0x7f53fc2f3170 {
      [Already dumped, see above]
    },
    {rxbuf_b, rxbuf_e} = {0x7f53fc2fc0b0, 0x7f53fc2fc74a},
    {pipeline_b, pipeline_e} = {(nil), (nil)},
    content_length = 0,
    body_status = none,
    first_byte_timeout = 0.000000,
    between_bytes_timeout = 0.000000,
  },
  http[req] = 0x7f53fc2f3218 {
    ws = (nil) {
    },
    hdrs {
    },
  },
  vdc = 0x7f53fc2fc028 {
    nxt = (nil),
    retval = 0,
  },
  flags = {
  },
  privs = 0x7f53fc2f3200 {
    MAGIC at 0x7f53fc2f3200 is 0x00000000 (Should be: VRT_PRIVS_MAGIC/0x03ba7502)
  },
  topreq = {
    req = 0x7f53fc2f3020 {
      [Already dumped, see above]
    },
  },
},
thr.busyobj = (nil) {
},
vmods = {
  std = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  blob = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  var = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  header = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  cookie = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  geoip2 = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  xkey = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  vsthrottle = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  directors = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  dynamic = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
},
```

Please help as this is clearing my cache every few days now.

_Originally posted by @jpastuszek in https://github.com/varnishcache/varnish-cache/issues/2572#issuecomment-516772773_

**Comment by jpastuszek** [comment]
Got another crash:
```
Panic at: Fri, 09 Aug 2019 11:38:27 GMT
Assert error in h2_rx_data(), http2/cache_http2_proto.c line 775:
  Condition((h2->mailcall) == 0) not true.
version = varnish-6.2.0 revision b14a3d38dbe918ad50d3838b11aa596f42179b54, vrt api = 9.0
ident = Linux,4.14.123-86.109.amzn1.x86_64,x86_64,-junix,-smalloc,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-hcritbit,epoll
now = 768454.170896 (mono), 1565350705.495268 (real)
Backtrace:
  0x43c6bb: /usr/sbin/varnishd() [0x43c6bb]
  0x49e102: /usr/sbin/varnishd(VAS_Fail+0x42) [0x49e102]
  0x46b41f: /usr/sbin/varnishd() [0x46b41f]
  0x46dec3: /usr/sbin/varnishd(h2_rxframe+0x463) [0x46dec3]
  0x46ffea: /usr/sbin/varnishd() [0x46ffea]
  0x45b743: /usr/sbin/varnishd() [0x45b743]
  0x45bc00: /usr/sbin/varnishd() [0x45bc00]
  0x7f9ff04dbde5: /lib64/libpthread.so.0(+0x7de5) [0x7f9ff04dbde5]
  0x7f9ff0204f1d: /lib64/libc.so.6(clone+0x6d) [0x7f9ff0204f1d]
thread = (cache-worker)
pthread.attr = {
  guard = 4096,
  stack_bottom = 0x7f9fe8d3d000,
  stack_top = 0x7f9fe8d4b000,
  stack_size = 57344,
}
thr.req = 0x7f9507e7f020 {
  vxid = 18696075, transport = H2
  step = 0x0,
  req_body = R_BODY_INIT,
  err_code = 1, err_reason = (null),
  restarts = 0, esi_level = 0,
  sp = 0x7f94d84dba20 {
    fd = 47, vxid = 18696075,
    t_open = 1565350705.491449,
    t_idle = 1565350705.494876,
    ws = 0x7f94d84dba60 {
      id = \"ses\",
      {s, f, r, e} = {0x7f94d84dba98, +224, (nil), +352},
    },
    transport = H2 {
      streams {
        0x00000000 idle
        0x00000001 open
      }
    }
    client = 154.66.183.11 54615 :2443,
  },
  ws = 0x7f9507e7f170 {
    id = \"req\",
    {s, f, r, e} = {0x7f9507e880b0, +9360, (nil), +61256},
  },
  http_conn = 0x7f9507e88050 {
    fd = 47 (@0x7f94d84dba44),
    doclose = NULL,
    ws = 0x7f9507e7f170 {
      [Already dumped, see above]
    },
    {rxbuf_b, rxbuf_e} = {0x7f9507e880b0, 0x7f9507e8a53f},
    {pipeline_b, pipeline_e} = {0x7f9507e88bdd, 0x7f9507e8a53f},
    content_length = 0,
    body_status = none,
    first_byte_timeout = 0.000000,
    between_bytes_timeout = 0.000000,
  },
  http[req] = 0x7f9507e7f218 {
    ws = (nil) {
    },
    hdrs {
    },
  },
  vdc = 0x7f9507e88028 {
    nxt = (nil),
    retval = 0,
  },
  flags = {
  },
  privs = 0x7f9507e7f200 {
    MAGIC at 0x7f9507e7f200 is 0x00000000 (Should be: VRT_PRIVS_MAGIC/0x03ba7502)
  },
  topreq = {
    req = 0x7f9507e7f020 {
      [Already dumped, see above]
    },
  },
},
thr.busyobj = (nil) {
},
vmods = {
  std = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  blob = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  var = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  header = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  cookie = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  geoip2 = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  xkey = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  vsthrottle = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  directors = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  dynamic = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
},
```

**Comment by jpastuszek** [comment]
I have patched Varnish 6.2.0 with this change but got another panic after few days. It tells me nothing so I don't know if this is even related to this problem but here it is:
```
Panic at: Fri, 04 Oct 2019 08:30:40 GMT
Assert error in Lck__Lock(), cache/cache_lck.c line 120:
  Condition((pthread_mutex_lock(&ilck->mtx)) == 0) not true.
version = varnish-6.2.0 revision b14a3d38dbe918ad50d3838b11aa596f42179b54, vrt api = 9.0
ident = Linux,4.14.123-86.109.amzn1.x86_64,x86_64,-junix,-smalloc,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-sfile,-hcritbit,epoll
now = 5595587.618954 (mono), 1570177838.943326 (real)
Backtrace:
  0x43c6bb: /usr/sbin/varnishd() [0x43c6bb]
  0x49e162: /usr/sbin/varnishd(VAS_Fail+0x42) [0x49e162]
  0x4370c9: /usr/sbin/varnishd(Lck__Lock+0x119) [0x4370c9]
  0x470090: /usr/sbin/varnishd() [0x470090]
  0x45b75b: /usr/sbin/varnishd() [0x45b75b]
  0x45bc30: /usr/sbin/varnishd() [0x45bc30]
  0x7f050ce84de5: /lib64/libpthread.so.0(+0x7de5) [0x7f050ce84de5]
  0x7f050cbadf1d: /lib64/libc.so.6(clone+0x6d) [0x7f050cbadf1d]
thread = (cache-worker)
pthread.attr = {
  guard = 4096,
  stack_bottom = 0x7f0501c3b000,
  stack_top = 0x7f0501c49000,
  stack_size = 57344,
}
thr.req = 0x7efa1a10d020 {
  vxid = 1844070, transport = H2
  step = 0x0,
  req_body = R_BODY_INIT,
  err_code = 1, err_reason = (null),
  restarts = 0, esi_level = 0,
  sp = 0x7efa3047bc20 {
    fd = 87, vxid = 1844070,
    t_open = 1570177535.691411,
    t_idle = 1570177838.942622,
    ws = 0x7efa3047bc60 {
      id = \"ses\",
      {s, f, r, e} = {0x7efa3047bc98, +224, (nil), +352},
    },
    transport = H2 {
      streams {
        0x00000000 idle
        0x0000004f c-rem
        0x0000005d closed
        0x00000069 open
      }
    }
    client = 217.168.138.54 51175 :2443,
  },
  ws = 0x7efa1a10d170 {
    id = \"req\",
    {s, f, r, e} = {0x7efa1a1160b0, +16400, (nil), +61256},
  },
  http_conn = 0x7efa1a116050 {
    fd = 87 (@0x7efa3047bc44),
    doclose = NULL,
    ws = 0x7efa1a10d170 {
      [Already dumped, see above]
    },
    {rxbuf_b, rxbuf_e} = {0x7efa1a1160b0, 0x7efa1a11a0b9},
    {pipeline_b, pipeline_e} = {0x7efa1a116bdd, 0x7efa1a11a0b9},
    content_length = 0,
    body_status = none,
    first_byte_timeout = 0.000000,
    between_bytes_timeout = 0.000000,
  },
  http[req] = 0x7efa1a10d218 {
    ws = (nil) {
    },
    hdrs {
    },
  },
  vdc = 0x7efa1a116028 {
    nxt = (nil),
    retval = 0,
  },
  flags = {
  },
  privs = 0x7efa1a10d200 {
    MAGIC at 0x7efa1a10d200 is 0x00000000 (Should be: VRT_PRIVS_MAGIC/0x03ba7502)
  },
  topreq = {
    req = 0x7efa1a10d020 {
      [Already dumped, see above]
    },
  },
},
thr.busyobj = (nil) {
},
vmods = {
  std = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  blob = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  var = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  header = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  cookie = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  geoip2 = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  xkey = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  vsthrottle = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 9.0},
  directors = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
  dynamic = {Varnish 6.2.0 b14a3d38dbe918ad50d3838b11aa596f42179b54, 0.0},
},
```

**Comment by Dridi** [comment]
This is a different panic, can you please open a new ticket? And ideally do you think you could give a try to the master branch? For example using weekly packages?
